### PR TITLE
feat: ✨ restore deployment feature parity

### PR DESCRIPTION
**Pull request by ecxyzzy**
Now we're cooking :fire:

Side note: While not entirely within scope, I've taken the liberty of migrating to [@tsconfig/node18](https://www.npmjs.com/package/@tsconfig/node18) and removing some entries from the root tsconfig's compilerOptions that were already enforced by ESLint anyways.

### CODE REVIEW

File: apps/api/bronya.config.ts [hunk 1]
```diff
@@ -141,25 +160,99 @@ class MyStack extends Stack {
...
+      stage = `staging-${process.env.PR_NUM}`;
+      break;
+    case "development":
+      stage = "dev";
+      break;
+    default:
+      throw new Error(
+        "Invalid NODE_ENV specified. Valid values are 'production', 'staging', and 'development'.",
+      );
+  }
```
Comment (ap0nia) [inline apps/api/bronya.config.ts:190]: Can we encapsulate this as a function defined at the top level of the module (or maybe in `@libs/common` :eyes:)

```suggestion
// at the top-level
function getStage() {
  switch (process.env.NODE_ENV) {
    case "production":
      return "prod";
    case "staging":
      if (!process.env.PR_NUM) {
        throw new Error("NODE_ENV was set to staging, but a PR number was not provided.");
      }
      return `staging-${process.env.PR_NUM}`;
    case "development":
      return "dev";
    default:
      throw new Error(
        "Invalid NODE_ENV specified. Valid values are 'production', 'staging', and 'development'.",
      );
  }
}

const stage = getStage()

```

File: apps/api/bronya.config.ts [hunk 2]
```diff
@@ -141,25 +160,99 @@ class MyStack extends Stack {
...
+      }),
+    );
+
+    result.api.methods.forEach((x) => {
+      try {
+        x.resource.addMethod("OPTIONS", optionsIntegration);
+      } catch {
+        // no-op
+      }
+    });
```
Comment (ap0nia) [inline apps/api/bronya.config.ts:250]: Nit
```suggestion
    result.api.methods.forEach((apiMethod) => {
      try {
        apiMethod.resource.addMethod("OPTIONS", optionsIntegration);
      } catch {
        // no-op
      }
    });
```

File: apps/api/bronya.config.ts [hunk 3]
```diff
@@ -141,25 +160,99 @@ class MyStack extends Stack {
...
   }
 
   return app;
 }
 
 if (isCdk()) {
-  main();
+  main().then();
```
Comment (ap0nia) [inline apps/api/bronya.config.ts:257]: Would this work alright too?

```suggestion
  await main()
```

File: tsconfig.json [hunk 4]
```diff
@@ -1,23 +1,16 @@
 {
+  "extends": "@tsconfig/node18/tsconfig.json",
```
Comment (ap0nia) [inline tsconfig.json:2]: What's the benefit of this? I've always written my tsconfigs from scratch, so maybe I've just been drinking the koolaid

**Review by ap0nia — APPROVED** [review]
Thanks, these changes look solid!

File: apps/api/bronya.config.ts (hunk 3 above)
Comment (ecxyzzy) [reply apps/api/bronya.config.ts:257]: Yeah that's fine. I keep forgetting we have top level await now lol

File: tsconfig.json (hunk 4 above)
Comment (ecxyzzy) [reply tsconfig.json:2]: Sensible defaults and fewer things for us to maintain? :shrug: I know Docusaurus and SvelteKit ship with their own tsconfigs at least, so I suppose there is some merit to it, but I guess that sounds koolaid-y.

**Comment by ecxyzzy** [comment]
Sike again looks like cors is broken :upside_down_face:

**Pull request merged**